Packs/CortexDataLake/Integrations/CortexDataLake/CortexDataLake.py

```diff
@@ -100,6 +97,102 @@ def _oproxy_authorize(self) -> Tuple[str, str, str, str, int]:
+
+        if first_failure_time and last_failure_time:
+            last_failure_datetime = datetime.fromtimestamp(last_failure_time)
+            first_failure_datetime = datetime.fromtimestamp(first_failure_time)
+            last_first_time_delta = last_failure_datetime - first_failure_datetime
+            now_last_time_delta = datetime.now() - last_failure_datetime
+
+            if last_first_time_delta < timedelta(hours=1):
+                if now_last_time_delta < timedelta(minutes=1):
+                    raise DemistoException(err_msg)
+            elif last_first_time_delta < timedelta(hours=48):
+                if now_last_time_delta < timedelta(minutes=10):
+                    raise DemistoException(err_msg)
+            else:
+                if now_last_time_delta < timedelta(minutes=60):
+                    raise DemistoException(err_msg)
+
+        return self._get_access_token()
+
+    def _get_access_token(self) -> dict:
+        """ Performs an http request to oproxy-cdl access token endpoint
+        In case of failure, handles the error, otherwise reset the failure counters and return the response
+
+        Returns: The oproxy response or raising a DemistoException
+
+        """
+        oproxy_response = self._http_request('POST',
+                                             '/cdl-token',
+                                             json_data={'token': get_encrypted(self.refresh_token, self.enc_key)},
+                                             error_handler=Client.error_handler,
+                                             timeout=(60 * 3, 60 * 3),
+                                             retries=3,
+                                             backoff_factor=10,
+                                             status_list_to_retry=[400])
+        self.reset_failure_times()
+        return oproxy_response
+
+    @staticmethod
+    def error_handler(oproxy_response: requests.Response):
+        """ Handles the oproxy response in case of a failure
+
+        Args:
+            oproxy_response: The oproxy response
+
+        """
+        integration_context = demisto.getIntegrationContext()
+        current_time = datetime.now().timestamp()
+        times_dict = {LAST_FAILURE_TIME_CONST: current_time}
+        if not integration_context.get(FIRST_FAILURE_TIME_CONST):
+            # first failure
+            times_dict[FIRST_FAILURE_TIME_CONST] = current_time
+        integration_context.update(times_dict)
+        demisto.setIntegrationContext(integration_context)
+
+        err_msg = f'Error in API call [{oproxy_response.status_code}] - {oproxy_response.reason}'
+        try:
+            # Try to parse json error response
+            error_entry = oproxy_response.json()
+            err_msg += f'\n{json.dumps(error_entry)}'
+            raise DemistoException(err_msg)
+        except ValueError:
+            err_msg += f'\n{oproxy_response.text}'
+            raise DemistoException(err_msg)
```

Comment:
What is this? 
Wasn't it easier just to add try except to the oproxy call?

Reply:
this is the error handler to be called in BaseClient's http_request in case of 400

Reply:
I think it is better to just add try-except to the proxy call if we catch an exception we check if it is 400, if it is then we'll save the failure time and raise the error.